BLUESPAWN-client/src/hunt/hunts/HuntT1100.cpp

```diff
@@ -1,116 +1,138 @@
+
+		if (aLevel == Aggressiveness::Cursory) {
+			asp_indicators.assign(R"(\bcmd.exe\b|\bpowershell.exe\b|\bwscript.shell\b|\bprocessstartinfo\b|createobject\("scripting.filesystemobject"\))");
+			jsp_indicators.assign(R"(\bcmd.exe\b|\bpowershell.exe\b)");
+		}
+		else if (aLevel == Aggressiveness::Normal) {
+			asp_indicators.assign(R"(\bcmd.exe\b|\bpowershell.exe\b|\bwscript.shell\b|\bprocessstartinfo\b|\bcreatenowindow\b|\bcmd\b|\beval request\b|\bexecute request\b|\boscriptnet\b|createobject\("scripting.filesystemobject"\))");
+			jsp_indicators.assign(R"(\bcmd.exe\b|\bpowershell.exe\b|\bgetruntime\(\)\.exec\b)");
+		}
+	}
+
+	void HuntT1100::AddDirectoryToSearch(const std::wstring& sFileName){
+		web_directories.emplace_back(sFileName);
+	}
+
+	void HuntT1100::AddFileExtensionToSearch(const std::wstring& sFileExtension) {
+		web_exts.emplace_back(sFileExtension);
+	}
+
+	int HuntT1100::ScanCursory(const Scope& scope, Reaction reaction){
+		LOG_INFO("Hunting for T1100 - Web Shells at level Cursory");
+		reaction.BeginHunt(GET_INFO());
+		SetRegexAggressivenessLevel(Aggressiveness::Cursory);
+
+		int identified = 0;
+
+		for (wstring path : web_directories) {
+			FileSystem::Folder  f = FileSystem::Folder((LPCWSTR)path.c_str());
+			FileSystem::FileSearchAttribs attribs;
+			attribs.extensions = web_exts;
+			std::vector<FileSystem::File> files = f.GetFiles(&attribs, -1);
+			for (auto& entry : files) {
+				long offset = 0;
+				long targetAmount = 1000000;
+				CHAR* read = (CHAR *)calloc(targetAmount + 1L, 1);
+				DWORD amountRead = 0;
+				wstring file_ext = entry.GetFileAttribs().extension;
+				do {
+					entry.Read(read, offset, targetAmount, amountRead);
+					read[amountRead] = '\0';
+					string sus_file(read);
+					transform(sus_file.begin(), sus_file.end(), sus_file.begin(), ::tolower);
+					if (file_ext.compare(L".php") == 0) {
+						if (regex_search(sus_file, match_index, php_vuln_functions)) {
+							LOG_ERROR("Located likely web shell in file " << WidestringToString(entry.GetFilePath()) << " in text " << sus_file.substr(match_index.position(), match_index.length()));
+						}
+					}
+					else if (file_ext.substr(0, 4).compare(L".jsp") == 0) {
+						if (regex_search(sus_file, match_index, jsp_indicators)) {
+							LOG_ERROR("Located likely web shell in file " << WidestringToString(entry.GetFilePath()) << " in text " << sus_file.substr(match_index.position(), match_index.length()));
+						}
+					}
+					else if (file_ext.substr(0, 3).compare(L".as") == 0) {
+						if (regex_search(sus_file, match_index, asp_indicators)) {
+							identified++;
+							LOG_ERROR("Located likely web shell in file " << WidestringToString(entry.GetFilePath()) << " in text " << sus_file.substr(match_index.position(), match_index.length()));
+						}
+					}
+					offset += amountRead - 1000;
+				} while (targetAmount <= amountRead);
+			}
+		}
+		reaction.EndHunt();
+		return identified;
+		return 0;

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consecutive return, break, continue, goto or throw statements are unnecessary.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=4959972)

Reply:
should really address this before merge @wtm99